internal/pkg/proc/proc.go

```diff
@@ -0,0 +1,83 @@
+package proc
+
+import (
+	"fmt"
+	"strings"
+
+	"code.cloudfoundry.org/bytefmt"
+	"github.com/prometheus/procfs"
+)
+
+type ProcessList struct {
+	Pid            int
+	PPID           int
+	NumThreads     int
+	VirtualMemory  uint64
+	ResidentMemory uint64
+	CPUTime        float64
+	State          string
+	Command        string
+	Executable     string
+	Args           string
+}
+
+// List processes the list of running processes and
+func List() ([]ProcessList, error) {
+	p, err := procfs.AllProcs()
+	if err != nil {
+		return nil, err
+	}
+
+	return format(p)
+}
+
+func format(procs procfs.Procs) ([]ProcessList, error) {
+	pl := make([]ProcessList, 0, len(procs))
+	var (
+		cmdline []string
+		comm    string
+		err     error
+		exe     string
+		stats   procfs.ProcStat
+	)
+
+	for _, proc := range procs {
+		comm, err = proc.Comm()
+		if err != nil {
+			return nil, err
```

Comment:
nitpicking here, but probably to handle `/proc` failures in the future, would it be better to have a function to format a single entry (which might fail), but overall when we grab the list, we can put there 'failed' entries? 

what I'm going to i that if some procs fail to be discovered, we skip them but return others. so that if something is broken for a single process, it doesn't break others